### DISCUSSION TITLE
Unity Deepgram connection closed.

### DISCUSSION THREAD

**Discussion by Itzsadlife** [deepgram/community#489] [General help]
Hello everyone, I am currently facing issue of connection close, and I do not know what the problem is. Can anyone help me troubleshoot this issue please. Thank you in advance.

OnMessage: {"type":"Metadata","transaction_key":"deprecated","request_id":"2b96da38-b989-44ba-bb4f-356878973d03","sha256":"incomplete","created":"2023-12-12T20:38:25.019Z","duration":29.41,"channels":1,"models":["96a295ec-6336-43d5-b1cb-1e48b5e6d9a4"],"model_info":{"96a295ec-6336-43d5-b1cb-1e48b5e6d9a4":{"name":"general","version":"2023-02-22.3","arch":"base"}}}
UnityEngine.Debug:Log (object)
DeepgramInstance:<Start>b__3_3 (byte[]) (at Assets/Script/DeepgramInstance.cs:63)
NativeWebSocket.WebSocket:DispatchMessageQueue () (at ./Library/PackageCache/com.endel.nativewebsocket@c5101c0776/WebSocket/WebSocket.cs:637)
DeepgramInstance:Update () (at Assets/Script/DeepgramInstance.cs:89)

Connection closed!
UnityEngine.Debug:Log (object)
DeepgramInstance/<>c:<Start>b__3_2 (NativeWebSocket.WebSocketCloseCode) (at Assets/Script/DeepgramInstance.cs:57)
NativeWebSocket.WebSocket/<Receive>d__36:MoveNext () (at ./Library/PackageCache/com.endel.nativewebsocket@c5101c0776/WebSocket/WebSocket.cs:700)
System.Runtime.CompilerServices.AsyncMethodBuilderCore/MoveNextRunner:Run ()
WaitForUpdate/MainThreadAwaiter:Complete () (at ./Library/PackageCache/com.endel.nativewebsocket@c5101c0776/WebSocket/WebSocket.cs:66)
WaitForUpdate/<CoroutineWrapper>d__4:MoveNext () (at ./Library/PackageCache/com.endel.nativewebsocket@c5101c0776/WebSocket/WebSocket.cs:78)
UnityEngine.SetupCoroutine:InvokeMoveNext (System.Collections.IEnumerator,intptr)

Another issue （does not show every time）
SocketException: An established connection was aborted by the software in your host machine.
Rethrow as IOException: Unable to read data from the transport connection: An established connection was aborted by the software in your host machine.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by nikolawhallon** [comment]
The response from Deepgram indicates that 29.41 seconds of audio was sent before the connection was closed, does this line up roughly with what you observed?

Can you share some of your code? The line numbers referenced in `DeepgramInstance.cs` don't appear to immediately match up with our example demo (https://github.com/deepgram/UnityDeepgramDemo/blob/main/Assets/DeepgramInstance.cs) so seeing what that looks like would help.

**Comment by KADURKA-ANIL** [comment]
@nikolawhallon  i am also facing the same issue where the deepgram is getting closed suddenly STT

deepgram STT: metadata received: {
  type: 'Metadata',
  transaction_key: 'deprecated',
  request_id: '865b577e-50ef-40ea-bd51-5006b5d3c129',
  sha256: 'incomplete',
  created: '2024-11-20T12:44:09.360Z',
  duration: 23.479937,
  channels: 1,
  models: [ '9c7ae805-e600-4e0f-a6a2-725be88b7ede' ],
  model_info: {
    '9c7ae805-e600-4e0f-a6a2-725be88b7ede': {
      name: '2-phonecall-nova',
      version: '2024-02-05.31606',
      arch: 'nova-2'
    }
  }
}
deepgram STT: disconnected

**Reply by nikolawhallon** [reply]
It looks like that connection closed after 23.5 seconds, did you receive transcripts during that time? And can you share some of your code?
